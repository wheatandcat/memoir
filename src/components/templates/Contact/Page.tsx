import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import * as Clipboard from 'expo-clipboard';
import { ConnectedType } from 'components/pages/Contact/Connected';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import { TextInput as RNTextInput } from 'react-native';
import useAutoFocusInput from 'hooks/useAutoFocusInput';

export type Props = ConnectedType & {};

const Page: React.FC<Props> = (props) => {
  const autoFocusProps = useAutoFocusInput(true, 500);

  const onCopyUserID = useCallback(() => {
    Clipboard.setString(props.userID);
  }, [props.userID]);

  const debug = false;

  return (
    <View style={styles.root}>
      {debug && (
        <>
          <View style={styles.title}>
            <Text color="secondaryLight" size="sm">
              ユーザーID
            </Text>
          </View>
          <View style={styles.userIDText}>
            <TouchableOpacity onPress={onCopyUserID}>
              <Text size="sm">{props.userID}</Text>
            </TouchableOpacity>
          </View>
        </>
      )}

      <View style={styles.title} pt={debug ? 4 : 0} pb={2}>
        <Text color="secondaryLight" size="sm">
          コメント
        </Text>
      </View>
      <RNTextInput
        multiline
        style={styles.inputText}
        autoCapitalize="none"
        onChangeText={props.onChangeText}
        maxLength={300}
        blurOnSubmit={false}
        {...autoFocusProps}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    paddingTop: theme().space(4),
  },
  title: {
    paddingVertical: theme().space(1),
    paddingHorizontal: theme().space(3),
  },
  userIDText: {
    backgroundColor: theme().color.background.light,
    paddingVertical: theme().space(3),
    paddingHorizontal: theme().space(3),
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderColor: theme().color.secondary.light,
  },
  inputText: {
    borderColor: theme().color.secondary.light,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    paddingTop: theme().space(2),
    paddingBottom: theme().space(1),
    paddingHorizontal: theme().space(3),
    height: 250,
    color: theme().color.secondary.main,
    backgroundColor: theme().color.background.light,
    lineHeight: 20,
    fontWeight: 'bold',
    textAlignVertical: 'top',
  },
});

export default memo(Page);
