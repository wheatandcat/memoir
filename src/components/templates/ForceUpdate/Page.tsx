import React, { memo, useCallback } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';
import AppLink from 'react-native-app-link';
import Button from 'components/atoms/Button';

export type Props = {};
const appStoreId = 1613532672;
const playStoreId = 'com.wheatandcat.memoir';

const Page: React.FC<Props> = () => {
  const onPress = useCallback(() => {
    AppLink.openInStore({
      appName: 'memoir',
      appStoreId,
      appStoreLocale: 'jp',
      playStoreId,
    });
  }, []);

  return (
    <View style={styles.root}>
      <View py={5} mx={3}>
        <Text lineHeight={30}>
          新しいバージョンがリリースされました。{'\n'}
          ストアからアプリのバージョンアップをして下さい。
        </Text>
      </View>

      <Button title="ストアへ" width={250} onPress={onPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    padding: theme().space(3),
    height: '100%',
    alignItems: 'center',
  },
});

export default memo(Page);
