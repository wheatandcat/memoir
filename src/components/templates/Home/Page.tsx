import React, { memo, useState, useCallback } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import DateInput from 'components/organisms/DateInput/DateInput';
import MemoirButton from 'components/molecules/Home/MemoirButton.tsx';
import Cards from 'components/organisms/Cards/Cards';
import SettingModal from 'components/organisms/SettingModal';
import AddItemModal from 'components/organisms/AddItemModal';
import { LinearGradient } from 'expo-linear-gradient';
import theme from 'config/theme';

dayjs.locale('ja');
dayjs.extend(advancedFormat);

type Props = {
  date: string;
  openSettingModal: boolean;
  onItem: () => void;
  onMemoir: () => void;
  onCloseSettingModal: () => void;
  onChangeDate: (date: string) => void;
};

type State = {
  openAddItemModal: boolean;
};

const initialState = (): State => ({
  openAddItemModal: false,
});

const Page: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>(initialState());

  const onAddItem = useCallback(() => {}, []);

  return (
    <LinearGradient
      // Background Linear Gradient
      colors={[theme().color.gradation[0], theme().color.gradation[1]]}
      style={styles.root}
    >
      <SettingModal
        isVisible={props.openSettingModal}
        onClose={props.onCloseSettingModal}
      />
      <AddItemModal
        isVisible={state.openAddItemModal}
        date={dayjs().format('YYYY-MM-DD')}
        onAdd={onAddItem}
        onClose={() => setState((s) => ({ ...s, openAddItemModal: false }))}
      />
      <ScrollView>
        <View style={styles.inner}>
          <DateInput date={props.date} onChange={props.onChangeDate} />
          <Cards
            onItem={props.onItem}
            onAddItem={() =>
              setState((s) => ({ ...s, openAddItemModal: true }))
            }
          />
        </View>
      </ScrollView>
      <MemoirButton onPress={props.onMemoir} />
    </LinearGradient>
  );
};

export default memo(Page);

const styles = StyleSheet.create({
  root: {
    height: '100%',
  },
  inner: {
    height: '100%',
  },
});
