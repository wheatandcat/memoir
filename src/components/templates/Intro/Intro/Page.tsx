import React, { memo } from 'react';
import { ConnectedType } from 'components/pages/Intro/Intro/Connected';
import IntroPanel from 'components/organisms/Intro/Intro';

export type Props = ConnectedType & {
  dayOfWeek: number;
  hours: number;
  minutes: number;
  notification: boolean;
};

const Intro: React.FC<Props> = (props) => {
  return (
    <IntroPanel
      dayOfWeek={props.dayOfWeek}
      hours={props.hours}
      minutes={props.minutes}
      notification={props.notification}
      onSaveNotification={props.onSaveNotification}
      onFinish={props.onFinish}
    />
  );
};

export default memo(Intro);
