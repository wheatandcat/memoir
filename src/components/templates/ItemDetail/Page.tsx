import React, { memo } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import View from 'components/atoms/View';
import CardDetail from 'components/organisms/CardDetail/CardDetail';
import InputDateWrap from 'components/organisms/InputDateWrap/InputDateWrap';
import theme from 'config/theme';
import { ItemQuery } from 'queries/api/index';

type Item = NonNullable<ItemQuery['item']>;

type Props = Pick<Item, 'categoryID' | 'title'> & {
  loading: boolean;
  date: string;
  onChangeDate: (date: string) => void;
};

const Page: React.FC<Props> = (props) => {
  return (
    <InputDateWrap
      date={props.date}
      onChangeDate={props.onChangeDate}
      firstItem
    >
      <ScrollView>
        <View style={styles.inner}>
          {props.loading && (
            <CardDetail
              date={props.date}
              title={props.title}
              categoryID={props.categoryID}
            />
          )}
        </View>
      </ScrollView>
    </InputDateWrap>
  );
};

export default memo(Page);

const styles = StyleSheet.create({
  inner: {
    height: '100%',
    marginHorizontal: theme().space(3),
    marginVertical: theme().space(4),
  },
});
