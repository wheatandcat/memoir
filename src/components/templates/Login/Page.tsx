import React, { memo } from 'react';
import { StyleSheet, Platform, TouchableOpacity } from 'react-native';
import theme from 'config/theme';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import Image from 'components/atoms/Image';
import * as AppleAuthentication from 'expo-apple-authentication';
import { UseFirebaseAuth } from 'hooks/useFirebaseAuth';

export type Props = {
  onAppleLogin: UseFirebaseAuth['onAppleLogin'];
  onGoogleLogin: UseFirebaseAuth['onGoogleLogin'];
};

const Page: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      {Platform.OS === 'ios' && (
        <AppleAuthentication.AppleAuthenticationButton
          buttonType={AppleAuthentication.AppleAuthenticationButtonType.SIGN_IN}
          buttonStyle={AppleAuthentication.AppleAuthenticationButtonStyle.BLACK}
          cornerRadius={5}
          style={styles.button}
          onPress={props.onAppleLogin}
        />
      )}
      {Platform.OS === 'android' && (
        <TouchableOpacity onPress={props.onGoogleLogin}>
          <View style={styles.googleButton}>
            <View>
              <Image
                source={require('../../../img/icon/icon_google.png')}
                width={25}
                height={25}
              />
            </View>
            <View px={2}>
              <Text fontFamily="NotoSansJP-Bold">Googleでログイン</Text>
            </View>
          </View>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
    padding: theme().space(4),
  },
  button: {
    width: 200,
    height: 44,
  },
  googleButton: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: StyleSheet.hairlineWidth,
    paddingHorizontal: theme().space(3),
    paddingVertical: theme().space(0),
    borderRadius: 10,
  },
});

export default memo(Page);
