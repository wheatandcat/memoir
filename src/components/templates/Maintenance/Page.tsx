import React, { memo, useEffect } from 'react';
import { StyleSheet } from 'react-native';
import { AppConfig } from 'lib/appConfig';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import theme from 'config/theme';

export type Props = AppConfig & {
  getMaintenance: () => void;
};

const Page: React.FC<Props> = (props) => {
  useEffect(() => {
    const interval = setInterval(() => {
      props.getMaintenance();
    }, 30000);

    return () => clearInterval(interval);
  }, [props]);

  return (
    <View style={styles.root}>
      <View py={5}>
        <Text textAlign="center">ただいまメンテンナンス中です。</Text>
      </View>

      {!!props.maintenancePeriod && (
        <View px={3}>
          <Text textAlign="center" color="secondaryLight" lineHeight={36}>
            【メンテンナンス日程】
          </Text>

          <Text textAlign="center" color="error">
            {props.maintenancePeriod}
          </Text>
        </View>
      )}

      <View px={3} pt={4}>
        <Text variants="small" color="secondaryLight">
          {props.maintenanceMessage}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
});

export default memo(Page);
