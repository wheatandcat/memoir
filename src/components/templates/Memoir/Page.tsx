import React, { memo } from 'react';
import { StyleSheet, Platform } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useNavigation } from '@react-navigation/native';
import Constants from 'expo-constants';
import View from 'components/atoms/View';
import DateCards from 'components/organisms/Memoir/DateCards';
import theme from 'config/theme';
import ShareButton from 'components/molecules/Memoir/ShareButton';
import { Props as PlainProps } from 'components/pages/Memoir/Plain';
import IconButton from 'components/molecules/IconButton';
import FocusAwareStatusBar from 'components/organisms/FocusAwareStatusBar';

export type Props = Pick<
  PlainProps,
  | 'items'
  | 'loading'
  | 'onLoadMore'
  | 'onItem'
  | 'pageInfo'
  | 'users'
  | 'search'
  | 'selectedUserIDList'
  | 'onScreenShot'
  | 'onChangeUserID'
> & {
  startDate: string;
  endDate: string;
};

const Page: React.FC<Props> = (props) => {
  const navigation = useNavigation();

  return (
    <SafeAreaView>
      <FocusAwareStatusBar
        backgroundColor={theme().color.background.main}
        style="dark"
      />
      <View style={styles.root}>
        <View style={styles.inner}>
          <DateCards
            startDate={props.startDate}
            endDate={props.endDate}
            items={props.items}
            pageInfo={props.pageInfo}
            loading={props.loading}
            onItem={props.onItem}
            onLoadMore={props.onLoadMore}
            users={props.users}
            search={props.search}
            selectedUserIDList={props.selectedUserIDList}
            onChangeUserID={props.onChangeUserID}
          />
        </View>
        <View style={styles.action}>
          <ShareButton
            onPress={props.onScreenShot}
            disabled={props.items.length === 0}
          />
        </View>
      </View>
      <View style={styles.close}>
        <IconButton
          name="highlight-off"
          size="base"
          onPress={() => navigation.goBack()}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  root: {
    height: '100%',
    backgroundColor: theme().color.background.main,
  },
  inner: {
    height: '100%',
  },
  action: {
    bottom: 0,
    height: Platform.OS === 'ios' ? 22 : 55,
    position: 'absolute',
  },
  close: {
    position: 'absolute',
    top: Constants.statusBarHeight,
    left: theme().space(2),
    backgroundColor: theme().color.background.main,
    borderRadius: 25,
  },
});

export default memo(Page);
