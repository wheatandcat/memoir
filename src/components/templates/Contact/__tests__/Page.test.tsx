import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Page, { Props } from '../Page';

const propsData = (): Props => ({
  userID: 'abc',
  loading: false,
  onContact: jest.fn(),
});

describe('components/templates/Contact/Page.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Page {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
