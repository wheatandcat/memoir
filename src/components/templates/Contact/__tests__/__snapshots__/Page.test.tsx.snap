// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/templates/Contact/Page.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "height": "100%",
      "paddingTop": 32,
    }
  }
>
  <View
    pb={2}
    pt={0}
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      color="secondaryLight"
      fontFamily="RobotoCondensed-Bold"
      size="sm"
      variants="body"
    >
      コメント
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoFocus={true}
    blurOnSubmit={false}
    maxLength={300}
    multiline={true}
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 0.5,
        "borderColor": "#5E5166",
        "borderTopWidth": 0.5,
        "color": "#362740",
        "fontWeight": "bold",
        "height": 250,
        "lineHeight": 20,
        "paddingBottom": 4,
        "paddingHorizontal": 16,
        "paddingTop": 8,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
